src/diffpy/utils/scattering_objects/diffraction_object.py

```diff
@@ -0,0 +1,492 @@
+import datetime
+from copy import deepcopy
+
+import numpy as np
+
+from diffpy.utils.tools import get_package_info
+
+QQUANTITIES = ["q"]
+ANGLEQUANTITIES = ["angle", "tth", "twotheta", "2theta"]
+DQUANTITIES = ["d", "dspace"]
+XQUANTITIES = ANGLEQUANTITIES + DQUANTITIES + QQUANTITIES
+XUNITS = ["degrees", "radians", "rad", "deg", "inv_angs", "inv_nm", "nm-1", "A-1"]
+
+x_grid_emsg = (
+    "objects are not on the same x-grid. You may add them using the self.add method "
+    "and specifying how to handle the mismatch."
+)
+
+
+class DiffractionObject:
+    """
+    A class to manipulate diffraction data, supporting basic arithmetic operations and
+    conversions between different diffraction data metrics such as Q, two-theta, and d-spacing.
+
+    Parameters
+    ----------
+    name : str, optional
```

Comment:
What is the difference between a parameter and an attribute here?

Also, per numpy standards, desriptions always start with "The" so wavelength would be "The wavelength of the radiation used in the diffraction experiment. Defaults to None." which, slightly magically, makes everything much more readable.  Please change everywhere.

Reply:
Fixed - added "The" in the front. Ref: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html